SYCL/ESIMD/api/functional/ctors/ctor_fill.hpp

```diff
@@ -0,0 +1,226 @@
+//===-- ctor_fill.hpp - Functions for tests on simd fill constructor
+//      definition. -------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides functions for tests on simd fill constructor.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "common.hpp"
+// For std::isnan
+#include <cmath>
+
+namespace esimd = sycl::ext::intel::experimental::esimd;
+namespace esimd_tests = esimd_test::api::functional::ctors;
+namespace esimd_functional = esimd_test::api::functional;
+
+// Descriptor class for the case of calling constructor in initializer context.
+struct initializer {
+  static std::string get_description() { return "initializer"; }
+
+  template <typename DataT, int NumElems>
+  static void call_simd_ctor(DataT init_value, DataT step, DataT *out) {
+    esimd::simd<DataT, NumElems> simd_by_init =
+        esimd::simd<DataT, NumElems>(init_value, step);
```

Comment:
```suggestion
    const auto simd_by_init =
        esimd::simd<DataT, NumElems>(init_value, step);
```
Minor, we might just avoid duplicating the same statement, as the type is strictly defined by the initializer itself

Reply:
This comments was applied [here](https://github.com/intel/llvm-test-suite/pull/598/commits/652dcd3b7c72dceecf81bb7bd486398d8664c24d#diff-7fc90b589428e6e89076f07f5ff280a217b7f5c8e836bfe6c742734825074ef4R31-R32)